{
  "number": 9352,
  "title": "Build successful when an error was logged",
  "body": "![image](https://github.com/dotnet/msbuild/assets/679326/dbf5bf45-0da4-4a0f-b3a3-db8f9758252f)\r\n\r\nBinlog available internally upon request.\r\n\r\nI've tried to isolate a standalone repro, but failed. Needs deeper than a cursory investigation. CallTarget may be involved. Static graph is enabled via /graph.",
  "state": "CLOSED",
  "createdAt": "2023-10-21T00:55:50Z",
  "updatedAt": "2025-05-28T18:04:14Z",
  "closedAt": "2025-05-28T18:04:12Z",
  "author": {
    "login": "KirillOsenkov"
  },
  "milestone": null,
  "assignees": {
    "nodes": [
      {
        "login": "maridematte"
      }
    ]
  },
  "labels": [
    "Area: Debuggability",
    "Area: Engine",
    "Area: Logging",
    "triaged"
  ],
  "comments": {
    "nodes": [
      {
        "body": "My gut feeling is that we need to log the targets still left on the stack here:\r\nhttps://github.com/dotnet/msbuild/blob/c36a54ed3308d1516ffe1a86b9086c42e4ca996f/src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs#L172-L176\r\n\r\nIf we log each target in this loop, binlog will be able to establish a causality of what chain has caused the error. Right now the information doesn't seem to be available in the binlog.",
        "createdAt": "2023-10-22T19:55:54Z",
        "author": {
          "login": "KirillOsenkov"
        }
      },
      {
        "body": "Related:\r\n * https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/641\r\n * https://github.com/dotnet/msbuild/issues/4936\r\n * https://github.com/dotnet/msbuild/issues/6528\r\n\r\nFYI @baronfel ",
        "createdAt": "2023-11-06T03:34:22Z",
        "author": {
          "login": "KirillOsenkov"
        }
      },
      {
        "body": "msbuild9352.binlog",
        "createdAt": "2023-11-06T03:36:07Z",
        "author": {
          "login": "KirillOsenkov"
        }
      },
      {
        "body": "This might have been fixed by: https://github.com/dotnet/msbuild/pull/9461. Let me know if that is the case since I cannot repro this issue locally.",
        "createdAt": "2025-05-28T09:38:46Z",
        "author": {
          "login": "maridematte"
        }
      },
      {
        "body": "OK let me close and if I see this again I can reopen.",
        "createdAt": "2025-05-28T18:04:12Z",
        "author": {
          "login": "KirillOsenkov"
        }
      }
    ]
  }
}